plugins/volumeFilter_test.go

```diff
@@ -0,0 +1,439 @@
+	db := connectTestDb()
+
+	emptyConfigVolumeQuery, e := makeDailyVolumeByDateQuery(db, "", "native", "native", []string{}, []string{})
+	if e != nil {
+		t.Log("could not make empty config volume query")
+		return
+	}
+
+	marketIDs := []string{"marketID"}
+	marketConfigVolumeQuery, e := makeDailyVolumeByDateQuery(db, "", "native", "native", []string{}, marketIDs)
+	if e != nil {
+		t.Log("could not make market config volume query")
+		return
+	}
+
+	accountIDs := []string{"accountID"}
+	accountConfigVolumeQuery, e := makeDailyVolumeByDateQuery(db, "", "native", "native", accountIDs, []string{})
+	if e != nil {
+		t.Log("could not make account config volume query")
+		return
+	}
+
+	accountMarketConfigVolumeQuery, e := makeDailyVolumeByDateQuery(db, "", "native", "native", accountIDs, marketIDs)
+	if e != nil {
+		t.Log("could not make account and market config volume query")
+		return
+	}
+
+	testCases := []struct {
+		name           string
+		configValue    string
+		exchangeName   string
+		tradingPair    *model.TradingPair
+		assetDisplayFn model.AssetDisplayFn
+		baseAsset      hProtocol.Asset
+		quoteAsset     hProtocol.Asset
+		db             *sql.DB
+		config         *VolumeFilterConfig
+
+		wantError        error
+		wantSubmitFilter SubmitFilter
+	}{
+		// Note: config can be additional market IDs, make sure to test additionalMarketIds and optionalAccountIDs
+		{
+			name:             "failure - display base",
+			configValue:      "",
+			exchangeName:     "",
+			tradingPair:      &model.TradingPair{Base: "FAIL", Quote: ""},
+			assetDisplayFn:   testAssetDisplayFn,
+			baseAsset:        hProtocol.Asset{},
+			quoteAsset:       hProtocol.Asset{},
+			db:               db,
+			config:           &VolumeFilterConfig{},
+			wantError:        fmt.Errorf("could not convert base asset (FAIL) from trading pair via the passed in assetDisplayFn: cannot recognize the asset FAIL"),
+			wantSubmitFilter: nil,
+		},
+		{
+			name:             "failure - display quote",
+			configValue:      "",
+			exchangeName:     "",
+			tradingPair:      &model.TradingPair{Base: "XLM", Quote: "FAIL"},
+			assetDisplayFn:   testAssetDisplayFn,
+			baseAsset:        testNativeAsset,
+			quoteAsset:       hProtocol.Asset{},
+			config:           &VolumeFilterConfig{},
+			wantError:        fmt.Errorf("could not convert quote asset (FAIL) from trading pair via the passed in assetDisplayFn: cannot recognize the asset FAIL"),
+			wantSubmitFilter: nil,
+		},
+		{
+			name:             "failure - query",
+			configValue:      "",
+			exchangeName:     "",
+			tradingPair:      &model.TradingPair{Base: "XLM", Quote: "XLM"},
+			assetDisplayFn:   testAssetDisplayFn,
+			baseAsset:        testNativeAsset,
+			quoteAsset:       testNativeAsset,
+			config:           &VolumeFilterConfig{},
+			db:               nil,
+			wantError:        fmt.Errorf("could not make daily volume by date Query: could not make daily volume by date action: the provided db should be non-nil"),
```

Comment:
this is a good test to have in general but since it's testing logic in the factory method for `queries.MakeDailyVolumeByDateForMarketIdsAction` it's best to leave out. It does test that we are propagating errors from subcomponents correctly, but that would result in too many tests and is better caught in other ways.

Reply:
That sounds good. Out of curiosity, though, how else would we catch propagating errors from subcomponents?